src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,260 @@
 import backend.resource.TurboIssue;
 import backend.resource.TurboLabel;
 import javafx.application.Platform;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
 import javafx.geometry.Insets;
+import javafx.scene.Node;
 import javafx.scene.control.*;
-import javafx.scene.input.KeyCode;
 import javafx.scene.layout.FlowPane;
 import javafx.scene.layout.VBox;
 import javafx.stage.Modality;
 import javafx.stage.Stage;
 
-import java.util.Collections;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.Optional;
+import java.util.ResourceBundle;
 
-public class LabelPickerDialog extends Dialog<List<String>> {
+import org.apache.logging.log4j.Logger;
+
+import ui.UI;
+import util.HTLog;
+
+/**
+ * Serves as a presenter that synchronizes changes in labels with dialog view  
+ */
+public class LabelPickerDialog extends Dialog<List<String>> implements Initializable {
 
-    private static final int VBOX_SPACING = 105; // seems like some magic number
     private static final int ELEMENT_MAX_WIDTH = 400;
+    private static final Insets GROUPLESS_PAD = new Insets(5, 0, 0, 0);
+    private static final Insets GROUP_PAD = new Insets(0, 0, 10, 10);
+    private static final Logger logger = HTLog.get(LabelPickerDialog.class);
+
+    /**
+     * This should contain the list of labels defined in the current repository
+     */
+    private final List<TurboLabel> repoLabels;
+    private final TurboIssue issue;
+    private LabelPickerState state;
 
-    private final LabelPickerUILogic uiLogic;
-    private final TextField textField;
-    private final FlowPane topPane;
-    private final VBox bottomBox;
+    private ChangeListener<String> listener;
```

Comment:
This field is never assigned, but it's used (removed from the text property). May be a bug.